clap_complete/src/shells/zsh.rs

```diff
@@ -110,12 +110,15 @@ _{bin_name_underscore}_commands() {{
     ret.push(parent_text);
 
     // Next we start looping through all the children, grandchildren, etc.
-    let mut all_subcommands = utils::all_subcommands(p);
+    let mut all_subcommand_bins: Vec<_> = utils::all_subcommands(p)
+        .into_iter()
+        .map(|(_sc_name, bin_name)| bin_name)
+        .collect();
 
-    all_subcommands.sort();
-    all_subcommands.dedup();
+    all_subcommand_bins.sort();
+    all_subcommand_bins.dedup();
 
-    for (_, ref bin_name) in &all_subcommands {
+    for bin_name in &all_subcommand_bins {
```

Comment:
> Commit 1. Fixing the iteration over all_subcommands in zsh.rs. We deduplicate values on (sc_name, bin_name) keys, but then only iterate on bin_name. This doesn't cause problems now, since all bin names seem to be unique. However, without fixing this, the second commit would have started generating duplicated functions with same names.

Please update the commit with this information, rather than exclusively leaving it to the PR

Also, could you go into more detail on the problem.  In what way are we getting duplicates with the other PR?

Reply:
Updated commit messages, thanks.

If I `git revert` the first commit, then the second one will generate this code:

```zsh
(( $+functions[_my-app__some_cmd_commands] )) ||
_my-app__some_cmd_commands() {
    local commands; commands=(
'sub_cmd:sub-subcommand' \
'help:Print this message or the help of the given subcommand(s)' \
    )
    _describe -t commands 'my-app some_cmd commands' commands "$@"
}
(( $+functions[_my-app__some_cmd_commands] )) ||
_my-app__some_cmd_commands() {
    local commands; commands=(
'sub_cmd:sub-subcommand' \
'help:Print this message or the help of the given subcommand(s)' \
    )
    _describe -t commands 'my-app some_cmd commands' commands "$@"
}
```

Which will work correctly (re-defining a function is okay), but we can fix this inefficiency.